code-challenges.test.js

```diff
@@ -16,30 +18,102 @@
 // --------------------1) Create a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
 
 // a) Create a test with an expect statement using the variable provided.
+//set up test that will describe when codedMessage gets called
+// if argument is secretCodeWord1 
+//actual result .toEqual("l4ck4d41s1c4l")
+// if argument is secretCodeWord2 
+//actual result .toEqual("g0bbl3dyg00k")
+
+describe('when codedMessage gets called',()=>{
+    it("if argument is secretCodeWord1, retrun l4ck4d41s1c4l",() =>{
+    const argument= secretCodeWord1;
+    const actualResult = codedMessage(argument)
+    expect(actualResult).toEqual("l4ck4d41s1c4l")
+    })
+    it("if argument is secretCodeWord2, retrun g0bbl3dyg00k",() =>{
+    const argument= secretCodeWord2
+    const actualResult = codedMessage(argument)
+    expect(actualResult).toEqual("g0bbl3dyg00k")
+    })
+})
+
+
+
+// // b) Create the function that makes the test pass.
+
+
+// //create a function that takes a string as an argument
+const codedMessage = (string) =>{
+// //strings.toLowerCase() and splits that string with ('') foreach of value and if value === 'a' return 4 elseif value value ==='e' return 3 elseif value === 'i' elseif value === 'o' return 0  else return value and returns that new array to a codeString.join("")
+
+    codedString = []
+    string.toLowerCase().split('').forEach((value) =>{
```

Comment:
consider using `.map()` instead of `.forEach()` since you're transforming an array and the expected length of the array will remain the same